Fix: Negative value when booksGoal < booksRead

oysters76 opened the pull request:
There was a bug that I identified when the booksGoal is lesser than the booksRead. 
The text that is displayed on the screen is a negative value. Therefore I did some changes so that a message is displayed. Currently, the message says 'You have completed the goal!' when such a situation occurs. 

I have added a new function called resetFrequencyDisplayTextParagraph, which brings the paragraph back to original status, 
because the message would modify the innerHTML of the paragraph tag. 
```javascript 
const resetFrequencyDisplayTextParagraph = () => {
    frequencyDisplayText.innerHTML = "You need to finish a book every "+
      "<span id='bookFrequencyDisplay'></span> days.";
      bookFrequencyDisplay = document.querySelector('#bookFrequencyDisplay');
}
```

Before change, 
![image](https://user-images.githubusercontent.com/75514064/104170586-6337ef80-5427-11eb-8d55-e33bbc84a6dd.png)
After change
![image](https://user-images.githubusercontent.com/75514064/104170630-764abf80-5427-11eb-97e6-24cd376be419.png)


Thanks

Commit:
Fix: Negative value when booksGoal < booksRead

A message is shown when the booksRemaining is negative.

cnissanka committed:
Change how the paragraph reset was done

It was earlier done using standard html, which is not very good for
code readability, therefore did it using createElement and createTextNode
method in JS.

cnissanka committed:
Add: Localstorage functionality to save the booksGoal variable

The booksGoal variable (if greater than 0) is stored in the localstorage
and is retrieved when the page loads.

Darknessflowers reviewed (changes requested):
Thank you for this! Can you please add the finished books to LocalStorage as well so both values are kept?

oysters76 commented:
Yes sure

cnissanka committed:
booksGoal and booksRead values added to localstorage

Whenever the booksGoal and the booksRead values are changed there are updated
in the localstorage as well as they are also retrieved upon refresh

oysters76 commented:
I have added a new commit which adds the booksRead to the localstorage as well

oysters76 commented on index.html:
```diff
@@ -31,7 +31,7 @@ <h1>Reading Tracker</h1>
           There are <span id="daysLeftDisplay"></span> days left in the year.
         </p>
-        <p>
+        <p id="frequencyDisplayText">
```
I added an id to this tag, so that it can be accessed in the JS code to write the message and also reset the tag

oysters76 commented on scripts.js:
```diff
@@ -6,21 +6,68 @@ let oneDay = 1000*60*60*24;
-console.log(currentDate);
+let frequencyDisplayText = document.querySelector('#frequencyDisplayText');
+
+const writeToLocalStorage = (goal, read) => {
```
This function writes to the localstorage, when the goal value and the read value are passed in. (This is only written if there is a Storage functionality in the browser)

oysters76 commented on scripts.js:
```diff
@@ -6,21 +6,68 @@ let oneDay = 1000*60*60*24;
+  return "";
+}
+
+const resetFrequencyDisplayTextParagraph = () => {
```
This function would reset the paragraph tag to its original structure, by using the createElement and createTextNode functions of JS
